frontend/rust-lib/flowy-document/src/manager.rs

```diff
@@ -138,21 +140,44 @@ impl DocumentManager {
       )
       .await?;
       let doc_state = encoded_collab.doc_state.to_vec();
-      let collab = self
-        .collab_for_document(
-          uid,
-          doc_id,
-          DataSource::DocStateV1(doc_state.clone()),
-          false,
-        )
-        .await?;
-      collab.lock().flush();
+      //FIXME: so apparently we need to create a full document twice in order to save a doc state?
```

Comment:
can use `Document::open_with` and then use this document instance to write doc_state to disk.